lib/mkContainer.nix

```diff
@@ -0,0 +1,35 @@
+# at this point just a thin wrapper around streamLayeredImage, but still worth
+# having to keep as much functionality as possible in lib rather than the module
+# system
+{
+  inputs,
+  lib,
+}: {
+  context,
+  system,
+  drv,
+  entrypoint ? null,
+  buildLayeredImageArgs,
+}: let
+  pkgs = context.nixpkgs.legacyPackages.${system};
+  name = buildLayeredImageArgs.name;
+  builderArgsHaveEntrypoint = buildLayeredImageArgs ? config.entrypoint && buildLayeredImageArgs.config.entrypoint != null;
+  runFloxActivate = entrypoint == null && !builderArgsHaveEntrypoint;
+in
+  pkgs.dockerTools.streamLayeredImage (buildLayeredImageArgs
+    // {
+      config = {
+        entrypoint =
+          if
+            entrypoint
+            != null
+          then
+            if builderArgsHaveEntrypoint
+            then throw "cannot specify both entrypoint and config.entrypoint"
+            else entrypoint
+          else if runFloxActivate
+          then ["${pkgs.bashInteractive}/bin/bash" "--rcfile" "${drv}/activate"]
+          else buildLayeredImageArgs.config.entrypoint;
```

Comment:
can this be moved into the module system? it'd be nice for the exposed config to be a complete source of truth for image creation, and to deduplicate this same default if we want to make using an external bundler possible

Reply:
I was trying to push logic into `lib` because in general the module system should contain as little logic as possible, and we should put as much as we can in `lib` functions, and I believe in general @tomberek prefers that. If we do decide to use an external bundler, couldn't we still expose the output of this function? E.g. `floxEnvs.name.containerConfig = lib.mkContainerConfig {...}`

Reply:
That makes sense, but the way I see it is this is basically just a default, which normally belong in the module system (and even if they won't fit well in options default, I see plenty of nixpkgs modules setting own config defaults under `config = { ... }`)

Reply:
@tomberek thoughts on this?

Reply:
updated the interface, we can move some of the logic about what the default entrypoint is into modules if we wish.